pkg/plugins/optional/helm/v1alpha/init.go

```diff
@@ -47,7 +47,15 @@ func (p *initSubcommand) InjectConfig(c config.Config) error {
 }
 
 func (p *initSubcommand) Scaffold(fs machinery.Filesystem) error {
-	scaffolder := scaffolds.NewInitHelmScaffolder(p.config, false)
+	ignoreFlags := map[string]bool{
+		"ignore-samples":       false,
+		"ignore-prometheus":    false,
+		"ignore-networkPolicy": false,
+		"ignore-certmanager":   false,
+		"ignore-webhook":       false,
+	}
```

Comment:
To make this work, we’d need to check all conditions explicitly instead of relying on scaffolding defaults.

For example, handling values isn’t as straightforward as it seems. We’d also need a way to validate and guarantee support across all options.

I know we opened an issue about this in the past, but I’m not sure how feasible or maintainable it is long-term.

If we go ahead with this, we should:
- Cover all relevant scaffold points
- Add CI tests to validate configs for each scenario
- Mock Helm charts and install them during CI to catch issues early

We could follow the same pattern used in our `e2e` tests: create mocks per case, deploy them, and verify outcomes. See `e2e/go/v4` and related GitHub Actions as reference.